System/Posix/ByteString/FilePath.hsc

```diff
@@ -50,7 +57,9 @@ import Data.Monoid ((<>))
 type RawFilePath = ByteString
 
 withFilePath :: RawFilePath -> (CString -> IO a) -> IO a
-withFilePath = useAsCString
+withFilePath path f = do
+  checkForInteriorNuls path
```

Comment:
Why are we converting the path to a CString twice, optimising for the failure case.  Why not assume success and use the `ptr` directly if the length check passes???

Reply:
Makes sense